classes/Services/DistributionApiService.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ */
+
+namespace PrestaShop\Module\AutoUpgrade\Services;
+
+use PrestaShop\Module\AutoUpgrade\Exception\DistributionApiException;
+
+class DistributionApiService
+{
+    public const API_URL = 'https://api.prestashop-project.org';
+
+    /**
+     * @throws DistributionApiException
+     */
+    public function getPhpVersionRequirements(string $targetVersion): array
+    {
+        $response = file_get_contents(self::API_URL . '/prestashop');
```

Comment:
> Errors/Exceptions[ ¶](https://www.php.net/manual/en/function.file-get-contents.php#refsect1-function.file-get-contents-errors)
>
> An [E_WARNING](https://www.php.net/manual/en/errorfunc.constants.php#constant.e-warning) level error is generated if filename cannot be found, length is less than zero, or if seeking to the specified offset in the stream fails.
>
> When file_get_contents() is called on a directory, an [E_WARNING](https://www.php.net/manual/en/errorfunc.constants.php#constant.e-warning) level error is generated on Windows, and as of PHP 7.4 on other operating systems as well.

Do we need to make these warnings silent with `@file_get_contents`?

Reply:
I think we can silence them